### PR TITLE
`mysql_virtual_network_rule` waitLRO

### PR DESCRIPTION
**Pull request by shu-ying789**
linked to #15641

### CODE REVIEW

File: internal/services/mysql/mysql_virtual_network_rule_resource.go [hunk 1]
```diff
@@ -90,27 +89,12 @@ func resourceMySQLVirtualNetworkRuleCreateUpdate(d *pluginsdk.ResourceData, meta
...
 
-	if _, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.ServerName, id.Name, parameters); err != nil {
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.ServerName, id.Name, parameters)
+	if err != nil {
 		return fmt.Errorf("creating/updating %s: %+v", id, err)
 	}
-
-	// Wait for the provisioning state to become ready
-	log.Printf("[DEBUG] Waiting for %s to become ready", id)
-	stateConf := &pluginsdk.StateChangeConf{
```
Comment (katbyte) [inline internal/services/mysql/mysql_virtual_network_rule_resource.go:99]: have you confirmed this is no longer required?

Comment (tombuildsstuff) [reply internal/services/mysql/mysql_virtual_network_rule_resource.go:99]: ping @shu-ying789 - have you confirmed that this is no longer required? the LRO itself didn't correctly wait for the ProvisioningState at one point in time.

Comment (ms-henglu) [reply internal/services/mysql/mysql_virtual_network_rule_resource.go:99]: Hi @tombuildsstuff , based on the tests, it seems that this is no longer required.

Comment (magodo) [reply internal/services/mysql/mysql_virtual_network_rule_resource.go:99]: I suggest we still keep the stateConf wait logic here, while only add up for the missing wait for future part in each of these PRs.

Comment (shu-ying789) [reply internal/services/mysql/mysql_virtual_network_rule_resource.go]: @tombuildsstuff stateConf I added it again, please check again, thanks!

**Review by katbyte — APPROVED** [review]
LGTM 🚜

**Pull request merged**